api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -48,6 +49,18 @@ default String name() {
    */
   TableScan newScan();
 
+  /**
+   * Create a new {@link TableScan scan} for this table with a {@link MetricsReporter metrics
+   * reporter}.
+   *
+   * <p>Once a table scan is created, it can be refined to project columns and filter data.
+   *
+   * @return a table scan for this table
+   */
+  default TableScan newScan(MetricsReporter metricsReporter) {
```

Comment:
Instead of modifying `Table`, we should probably extend `Scan` to accept any number of custom metrics report in addition to the default one. It should behave like a listener pattern.